testutil/contract.go

```diff
@@ -123,3 +146,80 @@ func CheckEthTxResponse(r abci.ResponseDeliverTx, cdc codec.Codec) (*evm.MsgEthe
 
 	return &res, nil
 }
+
+// CallContract is a helper function to call any arbitrary smart contract.
+func CallContract(ctx sdk.Context, evmosApp *app.Evmos, args ContractCallArgs) (res abci.ResponseDeliverTx, ethRes evm.MsgEthereumTxResponse, err error) {
+	var (
+		nonce    uint64
+		gasLimit = args.GasLimit
+		cdc      = evmosApp.AppCodec()
+		txData   sdk.TxMsgData
+	)
+
+	pk, ok := args.PrivKey.(*ethsecp256k1.PrivKey)
+	if !ok {
+		return res, ethRes, errors.New("error while casting type ethsecp256k1.PrivKey on provided private key")
+	}
+
+	key, err := pk.ToECDSA()
+	if err != nil {
+		return
+	}
+
+	addr := crypto.PubkeyToAddress(key.PublicKey)
+
+	if args.Nonce == nil {
+		nonce = evmosApp.EvmKeeper.GetNonce(ctx, addr)
+	} else {
+		nonce = args.Nonce.Uint64()
+	}
+
+	// if gas limit not provided
+	// use default
+	if args.GasLimit == 0 {
+		gasLimit = 1000000
+	}
+
+	// Create MsgEthereumTx that calls the contract
+	input, err := args.ABI.Pack(args.MethodName, args.Args...)
+	if err != nil {
+		return res, ethRes, fmt.Errorf("error while packing the input: %v", err)
+	}
+
+	msg := evm.NewTx(&evm.EvmTxArgs{
```

Comment:
nit: I would create a helper method for everything after this point—separate building/validating parameters and sending payload.

Reply:
done, added the `CheckEthTxResponse`